db/queries/recipe.sql

```diff
@@ -75,3 +75,70 @@ INSERT INTO recipes (
   initial_publish_date,
   forked_from,
   featured_revision;
+
+-- name: GetRecipeWithAuthorById :one
+SELECT
+  r.id AS recipe_id,
+  r.slug AS recipe_slug,
+  r.private AS recipe_private,
+  r.initial_publish_date AS recipe_initial_publish_date,
+  r.forked_from AS recipe_forked_from,
+  r.featured_revision AS recipe_featured_revision,
```

Comment:
I think rather than fetching the entire recipe here, we should just fetch the user but use the `recipe_id` for filtering. Take a look [here](https://github.com/forkd-app/forkd/blob/84fe5c0c692e07f3d6d0f3d5c229c5ed8e559a14/db/queries/revision.sql#L16) for an example of what I mean. This will then just return the `user` type so we don't have to do that switch on the type assertion mentioned above.